src/components/textarea/textarea.ts

```diff
@@ -0,0 +1,206 @@
+   */
+  @property()
+  rows = 4;
+
+  /**
+   * The validity message. Corresponds to `validity-message` attribute.
+   * If present and non-empty, this multi select shows the UI of its invalid state.
+   */
+  @property({ attribute: 'validity-message' })
+  validityMessage = '';
+
+  /**
+   * The value of the input.
+   */
+  @property()
+  set value(v: string) {
+    if (this.textarea) {
+      this.textarea.value = v.trim();
+    }
+  }
+
+  get value(): string {
+    if (this.textarea) {
+      return this.textarea.value;
+    }
+    return '';
+  }
+
+  /**
+   * Get a reference to the underlying textarea so we can directly apply values.
+   * This lets us fixe a bug where after a user would clear text, the value wouldn't update programmatically
+   */
+  @query('textarea')
+  protected textarea!: HTMLTextAreaElement;
+
+  /**
+   * handles updating the value when the non-named slot updates
+   *
+   * @param param0 the event - we pull the target from it since that's what we're interested in
+   */
+  protected _handleSlotChange({ target }: { target: HTMLSlotElement }) {
+    if (target.assignedNodes) {
+      this.value = target
+        .assignedNodes({ flatten: true })
+        .reduce((accumulator, node) => accumulator + (node ? node.textContent : ''), '');
+    }
+  }
+
+  render() {
+    const invalidIcon = WarningFilled16({ class: `${prefix}--text-area__invalid-icon` });
+
+    const textareaClasses = classnames(`${prefix}--text-area`, `${prefix}--text-area--v2`, {
+      [`${prefix}--text-area--invalid`]: this.invalid,
+    });
+
+    const labelClasses = classnames(`${prefix}--label`, {
+      [`${prefix}--label--disabled`]: this.disabled,
+    });
+
+    const helperTextClasses = classnames(`${prefix}--form__helper-text`, {
+      [`${prefix}--form__helper-text--disabled`]: this.disabled,
+    });
+
+    return html`
+      <label class="${labelClasses}" for="input">
+        <slot name="label-text">
+          ${this.labelText}
+        </slot>
+      </label>
+      <div class="${helperTextClasses}">
+        <slot name="helper-text">
+          ${this.helperText}
+        </slot>
+      </div>
+      <div class="${prefix}--text-area__wrapper" ?data-invalid="${this.invalid}">
+        ${this.invalid ? invalidIcon : null}
+        <textarea
+          ?autocomplete="${this.autocomplete}"
+          ?autofocus="${this.autofocus}"
+          class="${textareaClasses}"
+          cols="${this.cols}"
+          ?data-invalid="${this.invalid}"
+          ?disabled="${this.disabled}"
+          id="input"
+          name="${this.name}"
+          pattern="${this.pattern}"
+          placeholder="${this.placeholder}"
+          ?readonly="${this.readonly}"
+          ?required="${this.required}"
+          rows="${this.rows}"
+        ></textarea>
+        <!-- hidden slot to allow for slotted value changes like a native textarea -->
+        <slot class="slot-value" @slotchange="${this._handleSlotChange}"></slot>
```

Comment:
IIRC ShadyDOM shim does not support styling `<slot>`, given there is no corresponding element that's rendered in such shim environment. OTOH it's interesting that setting `textarea.innerHTML = '<span>foo</span>bar<span>baz</span>` with native `<textarea>` yields text node with that (non-parsed) HTML, instead of text node with `foobarbaz`.

Some ideas I can rattle off there is:

1. Using mutation observer instead of `onslotchange` and do `this._textarea.textContent = this.innerHTML` in the mutation observer callback
2. Make child nodes of `<bx-textarea>` no-op

Reply:
Yes I agree that 2. makes the component simpler and makes it more consistent with other form components - Thanks @cal-smith for your thoughts!